### PR TITLE
Tests and Fixes

### LINKED ISSUE #34: Check for Reentrancy in `mint`

### LINKED ISSUE #44: Mint failed and mint spot consumed
1. I gave myself two minting spots
2. I minted one token (I actually received an error on the frontend, didn't capture that error, but did receive the NFT)
3. I attempted to mint a second token, but I received a "No Spots" error

### LINKED ISSUE #46: Include ICP transaction fees in minting price

### PR DESCRIPTION
**Pull request by di-wu**
Closes #34 
Closes #44
Closes #46

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
     "make-admin": "npx ts-node -e 'require(\"./lib/keys.ts\").generateKey(\"admin\")'",
     "deploy:local": "dfx start --emulator --background --clean && ./.scripts/dfx-deploy",
     "test": "mocha --timeout 60000 -r ts-node/register 'tests/**/*.ts'",
-    "test:dfx": "npm run deploy:local && npm test && dfx stop"
+    "test:dfx": "npm run deploy:local && npm test && dfx stop",
+    "chaos": "CHAOS=true mocha --timeout 3600000 -r ts-node/register 'tests/chaos.test.ts'",
```
Comment (jorgenbuilder) [inline package.json:7]: <3

File: main.mo [hunk 2]
```diff
@@ -220,53 +223,47 @@ shared({caller}) actor class Rex(
...
         };
     };
 
-    /// The amount of principals that are currently minting...
-    /// It is basically a simple mutex lock?...
-    private var minting = 0;
-
     /// Returns how many principals are currently waiting on the mint endpoint.
-    public query func currentlyMinting() : async Nat { minting };
+    public query func currentlyMinting(token : Principal, index : Nat) : async Result.Result<Nat, Events.Error> {
```
Comment (jorgenbuilder) [inline main.mo:233]: Scopes the minting variable to a token, nice.

**Review by jorgenbuilder — APPROVED** [review]
Such test, much wow! Great fixes, thanks!

File: main.mo [hunk 3]
```diff
@@ -220,53 +223,47 @@ shared({caller}) actor class Rex(
...
-            // Revert call.
-            Events.Events.addSpot(events_, token, index, caller);
-        })) {
-            case (#ok(a))  a;
-            case (#err(e)) return #err(e);
+        data.minting += 1;
+        let revert = func () {
+            Events.Access.addSpot(data.access, caller);
+            data.minting -= 1;
         };
```
[the remark this answers is not in the document]
Comment (di-wu) [reply main.mo:256]: Yes, we should be safe to do so.

**Pull request merged**